apps/frontend/src/components/Reviews.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Container,
+  Grid,
+  TextField,
+  Typography,
+  IconButton,
+  FormHelperText,
+  Card,
+  CardContent,
+} from "@mui/material";
+import StarIcon from "@mui/icons-material/Star";
+import StarBorderIcon from "@mui/icons-material/StarBorder";
+import { useStyles } from "../styles/pageStyles";
+import _axios_instance from "../_axios_instance";
+import { colors } from "../styles/colors";
+
+export interface Review {
+  listing_review_id: string;
+  reviewerName: string;
+  stars: number;
+  comment: string;
+  userID: string;
+  listingID: string;
+  dateCreated: string;
+  dateModified: string;
+}
+
+const Reviews = ({ reviews }: { reviews: Review[] }) => {
+  const classes = useStyles();
+  const [reviewText, setReviewText] = useState("");
+  const [rating, setRating] = useState<number | null>(null);
+  const [reviewError, setReviewError] = useState("");
+
+  const handleReviewChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setReviewText(event.target.value);
+  };
+
+  const handleRatingChange = (newRating: number) => {
+    setRating(newRating);
+  };
+
+  const handlePostReview = async () => {
+    if (!rating || !reviewText) {
```

Comment:
I believe you're supposed to be able to rate without reviewing or review without rating, and some fields would be nullable

Reply:
Tbh, I made both mandatory because the listing cards look better with both lol. But I'll not make review text mandatory.
<img width="1125" alt="Screenshot 2024-06-22 at 10 00 54 PM" src="https://github.com/UVicMartletplace/martletplace/assets/72957712/5b87607d-7f20-4e66-b81f-25eaf7d66c59">